libs/sdk-ui-kit/src/@ui/UiMenu/components/defaults/DefaultUiMenuContentItem.tsx

```diff
@@ -0,0 +1,108 @@
+// (C) 2025 GoodData Corporation
+
+import React from "react";
+import cx from "classnames";
+import { e } from "../../menuBem.js";
+import { IUiMenuContentItemProps, IUiMenuContentItemWrapperProps, IUiMenuItemData } from "../../types.js";
+import { typedUiMenuContextStore } from "../../context.js";
+import { ShortenedText } from "../../../../ShortenedText/index.js";
+/**
+ * Default component for rendering content menu items.
+ * @internal
+ */
+export const DefaultUiMenuContentItemWrapper = React.memo(function DefaultUiMenuContentItemWrapper<
+    T extends IUiMenuItemData = object,
+>({ item }: IUiMenuContentItemWrapperProps<T>): React.ReactElement {
+    const { useContextStore, createSelector } = typedUiMenuContextStore<T>();
+    const selector = createSelector((ctx) => ({
+        onSelect: ctx.onSelect,
+        controlType: ctx.controlType,
+        scrollToView: ctx.scrollToView,
+        setFocusedId: ctx.setFocusedId,
+        makeItemId: ctx.makeItemId,
+        itemClassName: ctx.itemClassName,
+        ContentItemComponent: ctx.ContentItem,
+        isFocused: ctx.focusedItem?.id === item.id,
+    }));
+
+    const {
+        scrollToView,
+        controlType,
+        setFocusedId,
+        onSelect,
+        makeItemId,
+        itemClassName,
+        ContentItemComponent,
+        isFocused,
+    } = useContextStore(selector);
+
+    const scrollToItem = (element: HTMLLIElement | null) => {
+        if (!element || !isFocused) {
+            return;
+        }
+
+        scrollToView(element);
+    };
+
+    const handleMouseFocus = React.useCallback(() => {
+        if (controlType !== "mouse") {
+            return;
+        }
+        setFocusedId(item.id);
+    }, [controlType, item.id, setFocusedId]);
+
+    const handleSelect = React.useCallback(() => {
+        if (item.isDisabled) {
+            return;
+        }
+
+        onSelect(item);
+    }, [item, onSelect]);
+
+    const classNames = cx(
+        e("item-wrapper", {
+            isFocused,
+            isDisabled: !!item.isDisabled,
+        }),
+        typeof itemClassName === "function" ? itemClassName(item) : itemClassName,
+    );
+    return (
+        <li
+            ref={scrollToItem}
+            role="menuitem"
+            aria-haspopup="dialog" // TODO INE: make popup type configurable
```

Comment:
Forgotten TODO

Reply:
removed, roles will be handled in separate story